merp_picking_advanced_search/tests/test_picking_advanced_search.py

```diff
@@ -0,0 +1,100 @@
+            'location_dest_id': self.location_2.id,
+            'move_type': 'one',
+            'company_id': company.id,
+            'picking_type_id': picking_type.id
+        })
+        self.stock_picking_2 =self.env['stock.picking'].create({
+            'name': 'test_stock_picking_2',
+            'location_id': self.location_2.id,
+            'location_dest_id': self.location_1.id,
+            'move_type': 'one',
+            'company_id': company.id,
+            'picking_type_id': picking_type.id
+        })
+        self.stock_picking_3 = self.env['stock.picking'].create({
+            'name': 'test_stock_picking_3',
+            'location_id': self.location_2.id,
+            'location_dest_id': self.location_1.id,
+            'move_type': 'one',
+            'company_id': company.id,
+            'picking_type_id': picking_type.id
+        })
+        product_uom = company.currency_id
+        products = self.env['product.template'].search([], limit=3)
+        self.move_line_1 = self.env['stock.move.line'].create({
+            'picking_id': self.stock_picking_1.id,
+            'qty_done': 10.0,
+            'location_id': self.location_1.id,
+            'date': datetime.now(),
+            'location_dest_id': self.location_2.id,
+            'product_uom_qty': 20.0,
+            'product_uom_id': product_uom.id,
+            'product_id': products[0].id
+        })
+        self.move_line_2 = self.env['stock.move.line'].create({
+            'picking_id': self.stock_picking_2.id,
+            'qty_done': 25.0,
+            'location_id': self.location_2.id,
+            'date': datetime.now(),
+            'location_dest_id': self.location_1.id,
+            'product_uom_qty': 25.0,
+            'product_uom_id': product_uom.id,
+            'product_id': products[1].id
+        })
+        self.move_line_3 = self.env['stock.move.line'].create({
+            'picking_id': self.stock_picking_3.id,
+            'qty_done': 10.0,
+            'location_id': self.location_2.id,
+            'date': datetime.now(),
+            'location_dest_id': self.location_1.id,
+            'product_uom_qty': 10.0,
+            'product_uom_id': product_uom.id,
+            'product_id': products[2].id
+        })
+        self.move_line_4 = self.env['stock.move.line'].create({
+            'picking_id': self.stock_picking_3.id,
+            'qty_done': 10.0,
+            'location_id': self.location_1.id,
+            'date': datetime.now(),
+            'location_dest_id': self.location_2.id,
+            'product_uom_qty': 15.0,
+            'product_uom_id': product_uom.id,
+            'product_id': products[1].id
+        })
+
+    def test_check_product_not_moved(self):
+        stock_pickings = self.env['stock.picking'].browse([self.stock_picking_1.id, self.stock_picking_2.id, self.stock_picking_3.id])
+        for stock_picking in stock_pickings:
+            products_not_moved = self.compute_products_not_moved(stock_picking)
```

Comment:
I don't think its good to calculate what we already know and expect here. I think it would be more proper just to hardcode expected values here as they were given above (where you created moves).

@okuryan what do you think?